### ISSUE TITLE
IllegalIdentifierName doesn't check single no-parentheses lambda parameter

### ISSUE THREAD

**Issue by Powerrr** [checkstyle/checkstyle#9756]
Check documentation: https://checkstyle.sourceforge.io/config_naming.html#IllegalIdentifierName

```
/var/tmp$ javac Test.java 
/var/tmp$ 

/var/tmp$ cat config.xml 
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="IllegalIdentifierName"/>
    </module>
</module>

/var/tmp$ cat Test.java 
import java.util.function.Function;

public class Test {
  public static void main(String[] args) {
    Function<String, String> f1 = var -> var; // doesn't report violation
    Function<String, String> f2 = (var) -> var; // reports violation
  }
}

/var/tmp$ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp$ java $RUN_LOCALE -jar checkstyle-8.41.1-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /var/tmp/Test.java:6:36: Name 'var' must match pattern '(?i)^(?!(record|yield|var|permits|sealed|_)$).+$'. [IllegalIdentifierName]
Audit done.
Checkstyle ends with 1 errors.
```

---

Expected are two errors - one for each of the lambdas. But we only see one error.
Looks like when lambda has only one parameter and it's written without parentheses around it then it is not considered an identifier.

**Comment by nmancus1** [comment]
This is because `IllegalIdentifierName` does not check the `LAMBDA` token, but does check the `PARAMETER_DEF` token. Once we put parenthesis around `var` it becomes a CHILD OF `PARAMETER_DEF`, but without parenthesis,  it is simply a child of `LAMBDA`.:

```bash
    |       |   `--ASSIGN -> = [5:36]
    |       |       `--LAMBDA -> -> [5:42]
    |       |           |--IDENT -> var [5:38]
    |       |           `--EXPR -> EXPR [5:45]
    |       |               `--IDENT -> var [5:45]

```
VS 

```bash

    |       |   `--ASSIGN -> = [6:36]
    |       |       `--LAMBDA -> -> [6:44]
    |       |           |--LPAREN -> ( [6:38]
    |       |           |--PARAMETERS -> PARAMETERS [6:39]
    |       |           |   `--PARAMETER_DEF -> PARAMETER_DEF [6:39]
    |       |           |       |--MODIFIERS -> MODIFIERS [6:39]
    |       |           |       |--TYPE -> TYPE [6:39]
    |       |           |       `--IDENT -> var [6:39]
    |       |           |--RPAREN -> ) [6:42]
    |       |           `--EXPR -> EXPR [6:47]
    |       |               `--IDENT -> var [6:47]

```

 I can work on this once it is approved.

**Comment by strkkk** [comment]
It seems to me that check should support both cases, it is clearly a bug

**Comment by nmancus1** [comment]
I'm on it.

**Comment by rnveach** [comment]
Fix was merged
